starknet-client/src/types.rs

```diff
@@ -78,6 +84,28 @@ impl From<ExternalTransfer> for Vec<StarknetFieldElement> {
     }
 }
 
+/// A single party's state validity proof bundle used as input to the matching engine
+/// and forwarded to the contract for verification
+#[derive(Clone)]
+pub struct MatchPayload {
```

Comment:
nit: would feel nice to match the type definition in the contract, which has the `VALID COMMITMENTS` and `VALID REBLIND` statements as nested struct fields, matching the definitions of the `ValidCommitmentsStatement` and `ValidReblindStatement` types in the relayer, for which I've already implemented `CalldataSerializable` in the contracts e2e tests.

wonder if you have strong opinions on this? makes the code read a bit better and feel more modular

Reply:
The branch we're integrating with does not have this structure, we can add it to the relayer after the changes are made upstream: [see here](https://github.com/renegade-fi/renegade-contracts/blob/cairo2-no-poseidon-or-verifier/src/darkpool/types.cairo#L21)